websocket/connection/upgrader.go

```diff
@@ -0,0 +1,126 @@
+package connection
+
+import (
+	"fmt"
+	"net/http"
+	"raccoon/logger"
+	"raccoon/metrics"
+	pb "raccoon/websocket/proto"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/gorilla/websocket"
+)
+
+type Upgrader struct {
+	gorillaUg         websocket.Upgrader
+	Table             *Table
```

Comment:
One abstraction we can probably do here is to create a User interface that deals with handling user connection information and a table can be one implementation of that. This can help us to easily migrate to a DB in the future if required. Will leave this up to you though.

Reply:
Was thinking of using interface. Since its not public facing implementation and its unlikely we switch. Even if we need to switch its still easy to do without interface.